app/services/applications/change_funding_eligibility.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Applications
+  class ChangeFundingEligibility
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    OPTIONS = {
+      true => I18n.t("shared.yes"),
+      false => I18n.t("shared.no"),
+    }.freeze
+
+    attribute :application
+    attribute :eligible_for_funding, :boolean
+
+    validates :application, presence: true
+    validates :eligible_for_funding, inclusion: OPTIONS.keys
```

Comment:
I think we need to take into account `funded_place` and if it's set, as `funded_place` cannot be true if `eligible_for_funding` is false, worth confirming if its on accepted vs pending applications as `funded_place` is set on accepted applications from cohort 2024 onwards

Reply:
also worth checking declaration logic, if this is allowed on applications with declarations or not, if it is this becomes much more complex

Reply:
This appears to be the relevant logic in ECF, noting it only gets applied within the admin area so I've added it to the service rather than Application itself

* https://github.com/DFE-Digital/early-careers-framework/blob/main/app/models/npq_application.rb#L74-L75